Feature/item common details page

derveydylan opened the pull request.

derveydylan committed:
feat: item common details

Creates the page, with the different components.
Creates various ui components.
clsx is now used for conditional css classes.
Other changes.
(this is still wip, committing to do more commits with fewer changes in the future)

derveydylan committed:
feat: input file for images

Creates the beginning of the item commom form.
Creates a component to upload images and preview them.
Creates a util for file sizes.
Small changes to some components.

derveydylan committed:
feat: item common form

Creates generic input file component.
Various fixes.

derveydylan committed:
feat: exemplar edit

When clicking the edit button on a exemplar, it shows the exemplar form and prefills it with exemplar data.

derveydylan committed:
feat: input components imporvements

All input fields can be controlled or uncontrolled.
Errors about each field can be shown to the user.

derveydylan committed:
feat: data fetching separation

Item characteristics are now separated from the items, because they will be used in different contexts (when creating an item for example).

derveydylan committed:
feat: item and itemcommon forms

Forms are working correctly for creation and edition.
Updates items mocks to contain all data.
Adds suppliers mocks.
Updated some parts due to last commit.

derveydylan committed:
fix: multiselect works when uncontrolled

derveydylan committed:
fix: multiselect not having border

derveydylan committed:
feat: quick access options from Home page

Made the forms accessible by the meatballs menu of each object or exemplar on the Home page.
Various changes.

derveydylan committed:
fix: setTagColor called with undefined on mount

derveydylan committed:
docs: ui components

Also adds conditions to prevent displaying the component if the required props aren't provided.

derveydylan committed:
fix: components errors

derveydylan committed:
feat: combined ButtonLink and TextLink components

derveydylan commented:
Merge this branch after merging the home-page one (into develop too).

derveydylan committed:
fix: display item name

We now display the name of the object below the exemplar inventory number when viewing exemplar on the home page.

derveydylan committed:
fix: content overlapping with long inventory numbers

DidierViret committed:
Merge branch 'develop' into feature/item-common-details-page

DidierViret reviewed (commented):
From the homepage, clicking on the object's or specimen's image should have the same effect as clicking on it's name

DidierViret reviewed (commented):
When clicking on a specimen and displaying the object details view, the screen should automatically scroll to the clicked specimen (wich appears in yellow)

derveydylan committed:
feat: image in itemCommon is also a link

When clicking a exemplar, the exemplar also scrolls into view.

derveydylan committed:
feat: scroll to the exemplar

Scrolls to the exemplar when one is clicked from the home page.
Adds UseEffects comments.

derveydylan committed:
Merge branch 'feature/item-common-details-page' of https://github.com/OrifInformatique/stock_frontback into feature/item-common-details-page

derveydylan committed:
feat: various changes

In exemplars list of object details page, moved some buttons in a meatballs menu, the loan buttons can change dependning on the loan state, and the place where the item is loaned is displayed (i the item is loaned).
MeatballsMenu now accepts links.
Buttons aren't full rounded by default anymore.
Adds a function to teel a feature isn't developed yet (for dev purposes).
Other minor changes.

derveydylan commented:
All asked changes were made.

derveydylan committed:
fix: key prop to map missing in MeatballsMenu

Also adds display block (css class) in HTMLLink when styled as a button.

derveydylan committed:
fix: generate inv number button not having onClick function

derveydylan committed:
feat: using notDevelopedFeature dev utility method

Also notices a bug with multiselects.

DidierViret reviewed (commented):
When the item update form is displayed, the list of specimen can still be displayed but no action on these specimen should be visible. No buttons and no meatball menu.
This view is only intended to update an item, nothing else.

DidierViret reviewed (commented):
When the specimen update form is displayed, the item and other specimen informations can still be displayed but no action on them should be visible. No buttons and no meatball menu.
This view is only intended to update the selected specimen, nothing else.

DidierViret reviewed (commented):
When the add specimen form is displayed, the item and existing specimen informations can still be displayed but no action on them should be visible. No buttons and no meatball menu.
This view is only intended to add the new specimen, nothing else.

DidierViret reviewed (commented):
When the specimen update form is displayed, the specimen being modified should not appear a second time in the list of specimen. This could be confusing for the user.

DidierViret reviewed (commented):
When extended, the meatball menu should close when clicking outside of the menu. For now we have to click a second time on the meatball icon to close the menu.

derveydylan commented:
All requested changes have been made in the `feature/item-common-creation-page` branch (because it contains the more recent code).
